### DISCUSSION TITLE
Method to calculate ensemble pars at or near bnds?

### DISCUSSION THREAD

**Discussion by wkitlasten** [pypest/pyemu#456] [General]
I've been hacking away at my par ens to find reals with pars at/near bounds, which involves just reverting back to pandas via ._df. I'd hate to waste any more time if "there's a method for that..." Anything currently available?

**Comment by jtwhite79** [comment]
Ive been thinking about adding something like this for a while.  The bounds report in ies is smashing the realizations and all pars in a group together into a single container so it can obscure whats going.  I think you are in the right track with just some dataframe action.  The `ParameterEnsemble` class has those methods to enforce bounds, which have to detect parameters that are out of bounds (within given tolerance).  See:

[https://github.com/pypest/pyemu/blob/9b2b53bdcfbc3c94487b17b9cec3e63b0765de18/pyemu/en.py#L1647](https://github.com/pypest/pyemu/blob/9b2b53bdcfbc3c94487b17b9cec3e63b0765de18/pyemu/en.py#L1647)
